includes/class-amp-theme-support.php

```diff
@@ -1745,6 +1745,30 @@ public static function ensure_required_markup( DOMDocument $dom, $script_handles
 			$amp_scripts[ $missing_script_handle ] = AMP_DOM_Utils::create_node( $dom, 'script', $attrs );
 		}
 
+		// Remove scripts that had already been added but couldn't be detected from output buffering.
+		foreach ( array_diff( array_keys( $amp_scripts ), $script_handles ) as $superfluous_script_handle ) {
+			$src = $amp_scripts[ $superfluous_script_handle ]->getAttribute( 'src' );
+
+			// Skip scripts unrelated to AMP.
+			if ( ! $src || 0 !== strpos( $src, 'https://cdn.ampproject.org/' ) ) {
+				continue;
+			}
+
+			// Skip AMP runtime script.
+			if ( 'https://cdn.ampproject.org/v0.js' === $src ) {
+				continue;
+			}
+
+			$custom_element = $amp_scripts[ $superfluous_script_handle ]->getAttribute( 'custom-element' );
+
+			// Skip dynamic CSS classes script, it has no associated element.
+			if ( 'amp-dynamic-css-classes' === $custom_element ) {
+				continue;
+			}
```

Comment:
Instead of doing this single check here, and depending on what the AMP validator team comes back with regarding any list of components, perhaps `[ 'amp-dynamic-css-classes' ]` should be merged with `$script_handles` when it is passed into `array_diff()`. Then there would be no need for `$custom_element`, as the script handles are the component names.